### DISCUSSION TITLE
Does httpfs installation + S3 creds setting need to happen every time?

### DISCUSSION THREAD

**Discussion by ruchirj** [duckdb/duckdb#5985] [Q&A] [answered]
I am using duckdb JDBC and currently need to call the following snippet every time I want to read a parquet from S3:

```
    statement.executeUpdate("INSTALL httpfs")
    statement.executeUpdate("LOAD httpfs")
    statement.executeUpdate(s"SET s3_endpoint='$S3_ENDPOINT';")
    statement.executeUpdate(s"SET s3_access_key_id='$S3_ACCESS_KEY_ID';")
    statement.executeUpdate(s"SET s3_secret_access_key='$SECRET_ACCESS_KEY';")
    statement.executeUpdate(s"SET s3_url_style='$S3_URL_STYLE';")
    statement.executeUpdate(s"SET s3_region='$S3_REGION';")
```

Based on this: https://duckdb.org/docs/guides/import/s3_import.html I got the impression that I need to run these commands only once? However currently I need to set these on every statement. The other problem is you cannot use a statement concurrently and so everytime a file needs to be read these need to be run. Also this code snippet typically takes between 100-500ms and so is not exactly cheap.

**Comment by Mause** [comment]
Can you provide some working sample code that doesn't behave as expected?

**Comment by Mytherin** [comment, marked as answer]
`INSTALL` only needs to be run once in total. `LOAD` and setting parameters only need to be run once per connection. Are you perhaps instantiating a new connection for every statement?
